### ISSUE TITLE
`rotate` can flip the ascent and descent

### ISSUE THREAD

**Issue by florence** [racket/pict#12]
```
#lang racket
(require pict pict/shadow pict/conditional)
(pict-height
   (clip-descent
    (clip-ascent
     (rotate (text "sefsefse") pi))))
```

Produces

`-9.070312500000002`

This results in two observable errors:

```
> (blur
   (clip-descent
    (clip-ascent
     (rotate (text "sefsefse") pi)))
   3/2)
. . initialization for bitmap%: bad argument combination: 51 -6 #f #t
>  (clip-descent
    (clip-ascent
     (rotate (text "sefsefse") pi)))
. . ../../Applications/Racket/racket/collects/racket/contract/private/blame.rkt:156:0: set-clipping-rect: contract violation
  expected: (not/c negative?)
  given: -9.070312500000002
  in: an and/c case of
      the 4th argument of
      (->m
       real?
       real?
       (and/c real? (not/c negative?))
       (and/c real? (not/c negative?))
       void?)
  contract from: (class record-dc%)
  blaming: (unit language@)
   (assuming the contract is correct)
```

**Comment by florence** [comment]
This may actually be a bug with `clip-ascent` or `clip-descent`, because the ascent + descent of the rotated image is equal to the height, but clipping the ascent and descent results in a negative height:

```
-> (+ (pict-descent (rotate (text "sefsefse") pi)) (pict-ascent (rotate (text "sefsefse") pi)))
14.132812500000005
-> (pict-height (rotate (text "sefsefse") pi)))                                                                                                                                                14.132812500000005
-> (pict-height
      (clip-descent
       (clip-ascent
        (rotate (text "sefsefse") pi))))
-9.070312500000002
```

**Comment by florence** [comment]
Fixed by 457af69869b6af6de90fefe68b268925c477a8d0
